src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,80 @@
+package parser;
+
+import commands.AddCommand;
+import commands.ByeCommand;
+import commands.Command;
+import commands.DeleteCommand;
+import commands.DoneCommand;
+import commands.FindCommand;
+import commands.HelpCommand;
+import commands.ListCommand;
+import exceptions.DukeException;
+import tasks.Deadlines;
+import tasks.Events;
+import tasks.Todo;
+
+public class Parser {
+
+    /**
+     * Parses the input given and returns a command to be executed.
+     *
+     * @param line  TaskList object.
+     * @return The respective Command based on the input.
+     * @throws DukeException When the Index, Description or Details are empty based on the different Tasks.
+     */
+    public static Command parse(String line) throws DukeException {
```

Comment:
Is there a way you could modularize this further to avoid having a large block of code?

Reply:
Actually I was thinking about it, but so far, I haven't gotten around to making that large block of shorter yet. I believed the top half is already quite abstracted away, but the lower half of the method is still quite bloated indeed, so I'm seeing if I can abstract some of the details aways too. Noted with thanks!